lib/jxl/splines.cc

```diff
@@ -129,6 +129,8 @@ void ComputeSegments(const Spline::Point& center, const float intensity,
   segment.inv_sigma = 1.0f / sigma;
   segment.sigma_over_4_times_intensity = .25f * sigma * intensity;
   segment.maximum_distance = maximum_distance;
+  // TODO(eustas): this will work incorrectly for (center.y >= 1 << 23)
+  //               we have to use double precision in that case...
   ssize_t y0 = center.y - maximum_distance + .5f;
   ssize_t y1 = center.y + maximum_distance + 1.5f;  // one-past-the-end
   for (ssize_t y = std::max<ssize_t>(y0, 0); y < y1; y++) {
```

Comment:
In case `y1` is for whatever reason very large, this loop can cause a lot of memory consumption.

I don't think there is any reason to keep going after y >= frame_height, right? I think it's a good idea to replace the `y < y1` with `y < std::min<ssize_t>(y1, [FRAME HEIGHT]` (which will require some plumbing to actually know the frame height here).

Reply:
Overall, at some moment, we will have pipelined decoding and, hopefully, per-group spline rasterization. At that moment we need to remember to use group-origin relater coordinates, and viola - problem solved.

Reply:
ok, but can we for now cap `maximum_distance` and `y1` to something sensible? I think it is currently unbounded so in principle with the right input, I think you can make `y1` have arbitrary values here, which means arbitrary memory consumption.

(also `y1` could overflow causing no spline to be drawn, but I suppose that's a smaller problem)

Reply:
Done. Even better - limited spline area as well.